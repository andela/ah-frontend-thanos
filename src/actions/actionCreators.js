import ACTION_TYPES from './actionTypes';

export const SocialLogin = response => ({
  type: ACTION_TYPES.SOCIAL_LOGIN,
  payload: response,
});
export const LogIn = payload => ({
  type: ACTION_TYPES.LOGIN,
  payload,
});
export const socialLoginFailure = response => ({
  type: ACTION_TYPES.SOCIAL_LOGIN_FAILURE,
  payload: response,
});
export const testt = () => ({
  type: ACTION_TYPES.TEST,
  payload: 'test',
});
