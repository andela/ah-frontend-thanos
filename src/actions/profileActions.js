import axios from 'axios';
import { getProfile, getProfileError, editProfile } from './actionCreators';

export const getProfileAction = ({ username, token }) => (dispatch) => {
  axios.defaults.headers.common.Authorization = `Token ${token}`;
  return axios.get(`https://ah-backend-thanos-staging.herokuapp.com/api/profiles/${username}`, token)
    .then(response => dispatch(getProfile(response.data.results)))
    .catch(error => (
      dispatch(getProfileError({
        results: error.response.data.results,
        status_code: error.response.status,
      }))));
};

export const editProfileAction = (token, username, userProfile) => (dispatch) => {
  const profile = { profiles: userProfile };
  axios.defaults.headers.common.Authorization = `Token ${token}`;
  return axios.put(`https://ah-backend-thanos-staging.herokuapp.com/api/profiles/${username}`, profile)
    .then(response => dispatch(editProfile(response.data.results)))
    .catch((error) => {
      dispatch(getProfileError({
        results: error.response.data,
        status_code: error.response.status,
      }));
    });
};
