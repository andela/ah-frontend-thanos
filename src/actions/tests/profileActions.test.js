import moxios from 'moxios';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import ACTION_TYPE from '../actionTypes';
import { getProfile, getProfileError, editProfile } from '../actionCreators';
import { getProfileAction, editProfileAction } from '../profileActions';
import APP_URL from '../../utils/constants';

const mockStore = configureMockStore([thunk]);

describe('Test Profile actions', () => {
  let store;
  let url;
  let testUser;
  beforeEach(() => {
    moxios.install();
    store = mockStore({});
    testUser = 'johnDoe';
    url = `${APP_URL}/profiles/${testUser}`;
  });
  afterEach(() => {
    moxios.uninstall();
  });
  test('get/view Profile Action', () => {
    const profile = {
      username: 'rachaelminani',
    };
    expect(getProfile(profile)).toEqual(expect.objectContaining({
      type: ACTION_TYPE.GET_PROFILE_SUCCESS,
      payload: profile,
    }));
  });
  test(' Get profile action', () => {
    moxios.stubRequest(url, {
      status: 200,
      profile: { },
    });
    const expectedActions = [{ type: ACTION_TYPE.GET_PROFILE_SUCCESS }];
    const profile = {
      username: testUser,
      token: 'abcabc',
    };
    store.dispatch(getProfileAction(profile))
      .then(() => {
        expect(store.getActions()).toEqual(expectedActions);
      })
      .catch(() => {});
  });
  test(' Edit profile action passes', () => {
    moxios.stubRequest(url, {
      status: 200,
      response: {
        data: {
          results: {},
        },
      },
    });
    const expectedActions = [{ type: ACTION_TYPE.EDIT_PROFILE }];
    const profiles = {
      username: testUser,
      first_name: 'abcabc',
    };
    store.dispatch(editProfileAction('token', testUser, profiles))
      .then(() => {
        expect(store.getActions()).toEqual(expectedActions);
      });
  });

  test(' Edit profile action fails', () => {
    moxios.stubRequest(url, {
      status: 400,
    });
    const expectedActions = [{ payload: { results: undefined, status_code: 400 }, type: 'GET_PROFILE_ERROR' }];
    const profiles = {
      username: testUser,
      token: 'abcabc',
    };
    store.dispatch(editProfileAction('some-token', testUser, profiles))
      .then(() => {
        expect(store.getActions()).toEqual(expectedActions);
      });
  });
  it('LOGIN Action', () => {
    const profiles = {};
    expect(getProfileError(profiles)).toEqual(expect.objectContaining({
      type: ACTION_TYPE.GET_PROFILE_ERROR,
    }));
  });
  it('LOGIN Action', () => {
    const profiles = {};
    const freshProfile = editProfileAction(profiles, editProfile());
    expect(freshProfile).toMatchSnapshot();
  });
});
