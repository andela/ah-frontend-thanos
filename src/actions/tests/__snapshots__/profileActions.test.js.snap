// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Profile actions LOGIN Action 1`] = `[Function]`;
